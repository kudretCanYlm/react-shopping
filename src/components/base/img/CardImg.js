import React from 'react';
import PropTypes from 'prop-types';
import { DEFAULT_IMG_URL } from './common/common-props';

const CardImg = ({ imgUrl, className }) => {
  return (
    <div className={`flex-column card-img ${className}`}>
      <img src={imgUrl} />
    </div>
  );
};

//proptype
CardImg.propTypes = {
  imgUrl: PropTypes.string.isRequired,
  className: PropTypes.string
};

//default props
CardImg.defaultProps = {
  imgUrl: DEFAULT_IMG_URL,
  className: ''
};

export default CardImg;
