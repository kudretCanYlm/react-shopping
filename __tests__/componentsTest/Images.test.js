import React from 'react';
import Enzyme from 'enzyme';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
//componets
import CardImg from 'components/base/img/CardImg';

Enzyme.configure({ adapter: new Adapter() });

//varibles
const cardClassName = 'test classname';
const cardImgUrl = 'https://i.pinimg.com/originals/c7/42/20/c74220fcf2f83f4249b03ed2a2a87016.jpg';

describe('CardImgTest', () => {
  it('render correctly', () => {
    const component = renderer
      .create(<CardImg className={cardClassName} imgUrl={cardImgUrl} />)
      .toJSON();

    expect(component).toMatchSnapshot();
  });

  it('it has my className and imgurl', () => {
    const imgCom = shallow(<CardImg className={cardClassName} imgUrl={cardImgUrl} />);

    expect(imgCom.hasClass(cardClassName)).toEqual(true);
    expect(imgCom.html()).toContain(`src=\"${cardImgUrl}\"`);

    expect(imgCom.hasClass('not include ClassName')).toEqual(false);
    expect(imgCom.html()).not.toContain(`src=\"wrongurl\"`);
  });
});

describe('LogoImgTest', () => {});
