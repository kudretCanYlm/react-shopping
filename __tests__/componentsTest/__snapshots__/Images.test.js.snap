// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardImgTest render correctly 1`] = `
<div
  className="flex-column card-img test classname"
>
  <img
    src="https://i.pinimg.com/originals/c7/42/20/c74220fcf2f83f4249b03ed2a2a87016.jpg"
  />
</div>
`;
